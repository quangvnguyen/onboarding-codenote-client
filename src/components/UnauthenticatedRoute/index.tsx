import React from "react";
import { Route, Redirect } from "react-router-dom";
import { connect } from 'react-redux';

const mapStateToProps = (state) => ({
  isAuthenticated: state.authenticate.isAuthenticated,
});

const querystring = (name, url = window.location.href) => {
  name = name.replace(/[[]]/g, "\\$&");

  const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)", "i");
  const results = regex.exec(url);

  if (!results) {
    return null;
  }
  if (!results[2]) {
    return "";
  }

  return decodeURIComponent(results[2].replace(/\+/g, " "));
};

const UnauthenticatedRoute = ({ component: C, isAuthenticated, ...rest }) => {
  const redirect = querystring("redirect");
  return (
    <Route
      {...rest}
      render={props =>
        !isAuthenticated
          ? <C {...props} />
          : <Redirect
              to={redirect === "" || redirect === null ? "/" : redirect}
            />}
    />
  );
};

export default connect(mapStateToProps)(UnauthenticatedRoute);
