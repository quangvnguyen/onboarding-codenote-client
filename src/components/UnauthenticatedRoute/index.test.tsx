import React from 'react';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import UnauthenticatedRoute from './index';

const mockStore = configureMockStore();
window.location.href= '/notes';

describe("UnauthenticatedRoute Route component", () => {
  it('renders without crashing with authenticated', () => {
    const store = mockStore({
      authenticate: {
        isAuthenticated: true,
      }
    });
    const wrapper = mount(
      <Router>
        <UnauthenticatedRoute store={store} />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
