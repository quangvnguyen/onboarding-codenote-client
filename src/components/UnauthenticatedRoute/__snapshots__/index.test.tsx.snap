// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthenticatedRoute Route component renders without crashing with authenticated 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "7p6l0u",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(UnauthenticatedRoute)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <UnauthenticatedRoute
        dispatch={[Function]}
        isAuthenticated={true}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      >
        <Route
          dispatch={[Function]}
          render={[Function]}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          storeSubscription={
            Subscription {
              "listeners": Object {
                "clear": [Function],
                "get": [Function],
                "notify": [Function],
                "subscribe": [Function],
              },
              "onStateChange": [Function],
              "parentSub": undefined,
              "store": Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              },
              "unsubscribe": [Function],
            }
          }
        >
          <Redirect
            push={false}
            to="/"
          />
        </Route>
      </UnauthenticatedRoute>
    </Connect(UnauthenticatedRoute)>
  </Router>
</BrowserRouter>
`;
